src/foam/dao/jdbc/AbstractJDBCDAO.java

```diff
@@ -0,0 +1,283 @@
+  public FObject find_(X x, Object id) {
+    Connection c = null;
+    IndexedPreparedStatement stmt = null;
+    ResultSet resultSet = null;
+
+    try {
+      c = dataSource_.getConnection();
+      StringBuilder builder = threadLocalBuilder_.get()
+        .append("select * from ")
+        .append(tableName_)
+        .append(" where ")
+        .append(getPrimaryKey().createStatement())
+        .append(" = ?");
+
+      stmt = new IndexedPreparedStatement(c.prepareStatement(builder.toString()));
+      // TODO: add support for non-numbers
+      //stmt.setLong(((Number) o).longValue());
+      stmt.setObject(id);
+      resultSet = stmt.executeQuery();
+      if ( ! resultSet.next() ) {
+        // no rows
+        return null;
+        // In the doc, should be: throw new foam.dao.ObjectNotFoundException();
+      }
+
+      return createFObject(resultSet);
+    } catch (Throwable e) {
+      Logger logger = (Logger) x.get("logger");
+      logger.error(e);
+      return null;
+    } finally {
+      closeAllQuietly(resultSet, stmt);
+    }
+  }
+
+  @Override
+  public FObject remove_(X x, FObject obj) {
+    Connection c = null;
+    IndexedPreparedStatement stmt = null;
+
+    try {
+      c = dataSource_.getConnection();
+      StringBuilder builder = threadLocalBuilder_.get()
+        .append("delete from ")
+        .append(tableName_)
+        .append(" where ")
+        .append(getPrimaryKey().createStatement())
+        .append(" = ?");
+
+      stmt = new IndexedPreparedStatement(c.prepareStatement(builder.toString()));
+      // TODO: add support for non-numbers
+      //stmt.setLong(((Number) o.getProperty("id")).longValue());
+      stmt.setObject(obj.getProperty(getPrimaryKey().getName()));
+
+      int removed = stmt.executeUpdate();
+      if ( removed == 0 ) {
+        // throw new SQLException("Error while removing.");
+        // According to doc, no error when removing doesn't remove anything
+        return null;
+      }
+
+      return obj;
+    } catch (Throwable e) {
+      Logger logger = (Logger) x.get("logger");
+      logger.error(e);
+      return null;
+    } finally {
+      closeAllQuietly(null, stmt);
+    }
+  }
+
+  /**
+   * Create the table in the database and return true if it doesn't already exist otherwise it does nothing and returns false
+   * @param of ClassInfo
+   */
+  public abstract boolean maybeCreateTable(X x, ClassInfo of);
+
+  public AbstractJDBCDAO(X x, ClassInfo of) throws java.sql.SQLException, ClassNotFoundException {
+    setX(x);
+    setOf(of);
+
+    JDBCPooledDataSource jp = (JDBCPooledDataSource) x.get("JDBCDataSource");
+
+    dataSource_ = jp.getDataSource();
+
+    tableName_ = of.getObjClass().getSimpleName().toLowerCase();
+
+    getObjectProperties(of);
+
+    maybeCreateTable(x, of);
+
+  }
+
+  /**
+   * Returns list of properties of a metaclass
+   * @param of ClassInfo
+   */
+  protected void getObjectProperties(ClassInfo of){
+
+    if( properties_ == null ) {
```

Comment:
space after 'if'. The name of this method is misleading since it isn't a getter and returns void. Maybe something like buildPropertyList().